frontend/packages/schema-editor/src/components/SchemaInspector/ItemFieldsTab/ItemFieldsTable/ItemFieldsTableRow.tsx

```diff
@@ -0,0 +1,135 @@
+ * @property {boolean}[readonly] - If the field is readonly or not
+ * @property {function}[onEnterKeyPress] - Function to be executed on enter keypress
+ *
+ * @returns {ReactNode} - the rendered component
+ */
+export const ItemFieldsTableRow = ({
+  fieldNode,
+  readonly,
+  onEnterKeyPress,
+}: ItemFieldsTableRowProps): ReactNode => {
+  const { t } = useTranslation();
+  const { data, save } = useSchemaEditorAppContext();
+  const [isConfirmDeleteDialogOpen, setIsConfirmDeleteDialogOpen] = useState<boolean>();
+
+  const dispatch = useDispatch();
+
+  const fullPath = fieldNode.pointer;
+
+  const handleChangeNodeName = (newNodeName: string) => {
+    save(
+      setPropertyName(data, {
+        path: fullPath,
+        name: newNodeName,
+      }),
+    );
+  };
+
+  const onChangeType = (path: string, type: FieldType) => save(setType(data, { path, type }));
+
+  const onKeyDown = (e: KeyboardEvent<HTMLInputElement>) =>
+    e?.key === 'Enter' && onEnterKeyPress && onEnterKeyPress();
+
+  const changeRequiredHandler: ChangeEventHandler<HTMLInputElement> = (e) => {
+    save(
+      setRequired(data, {
+        path: fullPath,
+        required: e.target.checked,
+      }),
+    );
+  };
+
+  const deleteHandler = () => {
+    save(deleteNode(data, fullPath));
+    dispatch(removeSelection(fullPath));
+  };
+
+  return (
+    <tr>
+      <td className={cn(classes.tableColumnName, classes.tableCell)}>
+        <NameField
+          id={fieldNode.domId}
+          disabled={readonly}
+          handleSave={handleChangeNodeName}
+          onKeyDown={onKeyDown}
+          pointer={fullPath}
+          aria-label={t('schema_editor.field_name')}
+        />
+      </td>
+      <td className={cn(classes.tableColumnType, classes.tableCell)}>
+        <Select
+          hideLabel
+          inputId={`${fieldNode.domId}-typeselect`}
+          label={t('schema_editor.type')}
+          onChange={(fieldType) => onChangeType(fullPath, fieldType as FieldType)}
+          options={getTypeOptions(t)}
+          value={fieldNode.fieldType as FieldType}
+        />
+      </td>
+      <td className={cn(classes.tableColumnRequired, classes.tableCell)}>
+        <Center>
+          <Switch
+            size='small'
+            aria-label={t('schema_editor.required')}
+            checked={fieldNode?.isRequired ?? false}
+            disabled={readonly}
+            name='checkedArray'
+            onChange={changeRequiredHandler}
+          />
+        </Center>
+      </td>
+      <td className={cn(classes.tableColumnDelete, classes.tableCell)}>
+        <Center>
+          <AltinnConfirmDialog
+            open={isConfirmDeleteDialogOpen}
+            confirmText={t('schema_editor.datamodel_field_deletion_confirm')}
+            onConfirm={deleteHandler}
+            onClose={() => setIsConfirmDeleteDialogOpen(false)}
+            trigger={
+              <Button
+                aria-label={t('schema_editor.delete_field')}
```

Comment:
Her ville jeg heller brukt `title`. Da får man se teksten når man holder musen over.
```suggestion
                title={t('schema_editor.delete_field')}
```